lib/route.js

```diff
@@ -0,0 +1,18 @@
+SharedRoute = class {
+  constructor(router, pathDef, options) {
+    options = options || {};
+    
+    this.options = options;
+
+    this.name = options.name;
+
+    this.pathDef = pathDef;
+  
+    // Route.path is deprecated and will be removed in 3.0
+    this.path = this.pathDef;
+    
+    this._router = router;
```

Comment:
Since these are shared and used by sub class, let's remove the `_`.  Do this for all the other places.